internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -737,6 +765,28 @@ func partitionAndFormatDisks(logger *log.Logger, r runtime.Runtime) (err error)
 	}
 
 	for _, disk := range r.Config().Machine().Disks() {
+		var bd *blockdevice.BlockDevice
+
+		bd, err = blockdevice.Open(disk.Device, blockdevice.WithNewGPT(false))
+		if err != nil {
+			return err
+		}
+
+		// nolint: errcheck
+		defer bd.Close()
+
+		var pt table.PartitionTable
+
+		pt, err = bd.PartitionTable(true)
+		if err != nil {
+			return err
+		}
+
+		if len(pt.Partitions()) > 0 {
+			logger.Printf(("skipping setup of %q, found existing partitions"), disk.Device)
+			continue
+		}
+
```

Comment:
Improved detection of existing partitions, which is critical so that we don't repartition/reformat any extra disks.